turbinia/workers/artifact.py

```diff
@@ -0,0 +1,80 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task for running Plaso."""
+
+from __future__ import unicode_literals
+
+import os
+
+from turbinia import config
+from turbinia.evidence import ExportedFileArtifact
+from turbinia.workers import TurbiniaTask
+
+
+class FileArtifactExtractionTask(TurbiniaTask):
+  """Task to run image_export (log2timeline)."""
+
+  def __init__(self, artifact_name):
+    super(FileArtifactExtractionTask, self).__init__()
+    self.artifact_name = artifact_name
+
+  def run(self, evidence, result):
+    """Task that process data with Plaso.
+
+    Args:
+        evidence: Path to data to process.
+        result: A TurbiniaTaskResult object to place task results into.
+
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    config.LoadConfig()
+
+    export_directory = os.path.join(self.output_dir, 'export')
+    image_export_log = os.path.join(self.output_dir, '{0:s}.log'.format(self.id))
+
+    cmd = [
+        'image_export.py',
+        '--logfile',
+        image_export_log,
+        '-w',
+        export_directory,
+        '--artifact_filters',
+        self.artifact_name,
+        ]
+    if config.DEBUG_TASKS:
+      cmd.append('-d')
+
+    # Path to the source image/directory.
+    cmd.append(evidence.local_path)
+
+    result.log('Running image_export as [{0:s}]'.format(' '.join(cmd)))
+
+    ret, _ = self.execute(cmd, result, save_files=[image_export_log])
+    if ret:
+      result.close(False, "image_export.py failed.")
+
+    for dirpath, _, filenames in os.walk(export_directory):
+      for filename in filenames:
+        exported_artifact = ExportedFileArtifact(
+            artifact_name=self.artifact_name)
+        exported_artifact.local_path = os.path.join(dirpath, filename)
+        # exported_artifact.copyable = True
+        result.log('Adding artifact {0:s}'.format(filename))
+        result.add_evidence(exported_artifact, evidence.config)
```

Comment:
I'm not sold on the idea to emit one evidence file per artifact. What if I need multiple files extracted to do my analysis.. how will I make sure they all end up in my analysis job and not at any other worker who picks up the evidence? This is a larger discussion though, and I'm totally fine to leave this be and discuss that offline.

Reply:
Don't all workers get all the of the evidence types they want? If some other workers get this file, that seems WAI to me.

I think it might be useful to have different sorts of artifact extractors, some that grab a single artifact, and others that want some collection.